ipatests/test_integration/test_external_ca.py

```diff
@@ -218,3 +219,66 @@ def test_external_ca_dirsrv_stop(self):
         tasks.kinit_admin(self.master)
         result = self.master.run_command(['ipa', 'user-show', 'admin'])
         assert 'User login: admin' in result.stdout_text
+
+
+class TestExternalCAInvalidCert(IntegrationTest):
+    """Manual renew external CA cert with invalid file"""
+
+    def test_external_ca(self):
+        # Step 1 of ipa-server-install.
+        result = install_server_external_ca_step1(self.master)
+        assert result.returncode == 0
+
+        # Sign CA, transport it to the host and get ipa a root ca paths.
+        root_ca_fname, ipa_ca_fname = tasks.sign_ca_and_transport(
+            self.master, paths.ROOT_IPA_CSR, ROOT_CA, IPA_CA)
+
+        # Step 2 of ipa-server-install.
+        result = install_server_external_ca_step2(
+            self.master, ipa_ca_fname, root_ca_fname)
+        assert result.returncode == 0
+
+        result = self.master.run_command([paths.IPA_CACERT_MANAGE, 'renew',
+                                          '--external-ca'])
+        assert result.returncode == 0
+        result = self.master.run_command(['grep', '-v', 'CERTIFICATE', ipa_ca_fname])
+        contents = result.stdout_text
+        invalid_cert = os.path.join(self.master.config.test_dir, BAD_CERT)
+        self.master.run_command(['touch', invalid_cert])
+        self.master.put_file_contents(invalid_cert, contents)
+        # Sign CA, transport it to the host and get ipa a root ca paths.
+        root_ca_fname, ipa_ca_fname = tasks.sign_ca_and_transport(
+        self.master, paths.IPA_CA_CSR, ROOT_CA, IPA_CA)
+        # renew CA with invalid cert
+        result = self.master.run_command([paths.IPA_CACERT_MANAGE, 'renew',
```

Comment:
You can make the code more readable with a different indentation and argument passing style.

```
        result = self.master.run_command(
            [paths.IPA_CACERT_MANAGE, 'renew',
             '--external-cert-file', invalid_cert,
             '--external-cert-file={}', root_ca_fname],
            raiseonerr=False
        )
```